crates/typos-cli/src/file_type_specifics.rs

```diff
@@ -0,0 +1,66 @@
+//! This module specifies [`EngineConfig`] defaults for the file types defined in [`default_types`].
+//!
+//! [`EngineConfig`]: crate::config::EngineConfig
+//! [`default_types`]: crate::default_types
+
+/// Set `check_file` to `false` for these types.
+pub const NO_CHECK_TYPES: &[&str] = &["cert", "lock"];
+
+pub const TYPE_SPECIFIC_DICTS: &[(&str, StaticDictConfig)] = &[
+    (
+        "py",
+        StaticDictConfig {
+            ignore_idents: &["NDArray"],
+            ignore_words: &[],
+        },
+    ),
+    (
+        "rust",
+        StaticDictConfig {
+            ignore_idents: &["flate2"],
+            ignore_words: &["ser"],
+        },
+    ),
+    (
+        "vim",
+        StaticDictConfig {
+            ignore_idents: &["windo"],
+            ignore_words: &[],
+        },
+    ),
+    (
+        "vimscript",
+        StaticDictConfig {
+            ignore_idents: &["windo"],
+            ignore_words: &[],
+        },
+    ),
+];
+
+pub struct StaticDictConfig {
+    pub ignore_idents: &'static [&'static str],
+    pub ignore_words: &'static [&'static str],
+}
+
+#[cfg(test)]
+#[test]
+fn test_type_specific_dicts() {
+    let types: Vec<_> = TYPE_SPECIFIC_DICTS.iter().map(|(typ, _)| typ).collect();
+    for types in types.as_slice().windows(2) {
```

Comment:
If you sort the list and render it into a newline separated string, you could get a nice view of the whole thing at once with `snapshot::assert_eq`

Reply:
Yeah ... but I don't really see the advantage of that. The current code is already simple and produces nice error messages.

Reply:
I disagree about the current code being simple

Reply:
Ok, point taken. I agree that with `.join("\n")` and `snapbox::assert_eq` the test code would become simpler. However the error messages would as I see it become worse. Compare:

```
thread 'file_type_specifics::test_type_specific_dicts' panicked at 'Eq failed: 
--- Expected
+++ Actual
   1    1 | py
        2 + rust
   2    3 | vim
   3      - rust
   4    4 | vimscript∅
```

with:

> thread 'file_type_specifics::test_type_specific_dicts' panicked at '"rust" should come before "vim" in TYPE_SPECIFIC_DICTS', crates/typos-cli/src/file_type_specifics.rs:65:9

Note how the current error message mentions `TYPE_SPECIFIC_DICTS`. Unless I am missing something [snapbox::Assert](https://docs.rs/snapbox/latest/snapbox/struct.Assert.html) does not appear to support customized error messages.

Reply:
I think its good enough if `test_type_specific_dicts` is renamed to `test_type_specific_dicts_is_sorted`

Reply:
The test does however not only check if the slice is sorted but also that it does not contain duplicate entries.

My current code provides a different error message depending on whether the order is wrong or a duplicate entry exists, which I think is nice.

Yes we could have two test functions with largely the same code but at that point I'd argue that my current code is more elegant.

Reply:
In a lot of cases, I try to defer to the implementer as they are the ones who is immediately making a change.  For example, I had overlooked the de-duping in the name.

However, in the case of how this is handled, what I do not want to see
- Manual sorting checks
- Manual de-duping checks

And ideally, we don't show people a pigeon-hole view of the problem but that they have to fix one step at a time but we show them the whole set.  There are ways my suggestion can work between how the function is named and the naming of the variabled passed to the assert.  Alternatively, I'm fine with something like [pretty_assertions](https://crates.io/crates/pretty_assertions) being used.

So please change this!

Reply:
Thanks for your attention to detail and elaborating your reasoning. That does make a lot of sense. I split the test in two (which I now realize is better anyway) and updated it to use `snapbox::assert_eq` and `.join("\n")` like you initially suggested.